pkgs/dart_flutter_team_lints/lib/analysis_options.yaml

```diff
@@ -18,13 +18,15 @@ include: package:lints/recommended.yaml
 analyzer:
   language:
     strict-casts: true
+    strict-inference: true
 
 linter:
   rules:
     # consistency
     - avoid_empty_else
     - avoid_shadowing_type_parameters
     - avoid_types_as_parameter_names
+    - avoid_unused_constructor_parameters
```

Comment:
This one could lead to forced breaking API changes if a constructor parameter stops being used but is still passed by callers.

I don't know if it would ever occur that a constructor argument starts being ignored and it's not already a breaking behavior change, but I'm a little paranoid about lints that could suggest breaking API changes because an author applying a quick fix is less likely to consider impact on usages than an author manually changing arguments...

Reply:
Yes, people will feel the need to address these issues, and will assume that the lint is telling them the right thing to do.